src/Microsoft.AspNet.Mvc.ModelBinding/Validation/AssociatedValidatorProvider.cs

```diff
@@ -38,16 +39,15 @@ private IEnumerable<IModelValidator> GetValidatorsForProperty(ModelMetadata meta
                     "metadata");
             }
 
-            var attributes = property.GetCustomAttributes();
-            return GetValidators(metadata, attributes);
+            var attributes = ModelAttributes.GetAttributesForProperty(metadata.ContainerType, property);
+            return GetValidators(metadata, attributes.OfType<Attribute>());
```

Comment:
consider changing the signature of `GetValidators` instead. It's likely that nothing in that method depends on the API surface of `Attribute`